eth/gasprice/scroll_gasprice.go

```diff
@@ -43,6 +44,15 @@ func (oracle *Oracle) calculateSuggestPriorityFee(ctx context.Context, header *t
 	// capacity margin
 	receipts, err := oracle.backend.GetReceipts(ctx, header.Hash())
 	if receipts == nil || err != nil {
+		// Handle context cancellation gracefully
+		if errors.Is(err, context.Canceled) {
+			log.Debug("gas price calculation cancelled due to context cancellation", "block number", header.Number)
+			// Return cached values if available, otherwise use defaults
+			if lastIsCongested {
+				return lastPrice, lastIsCongested
+			}
+			return suggestion, isCongested
+		}
```

Comment:
Should we combine this error out with the following one, the logic is quite similar. And
```
log.Debug("failed to get block receipts during calculating suggest priority fee", "block number", header.Number, "err", err)
```
seems to be suited for both cases.